crates/runtime/src/logic.rs

```diff
@@ -108,6 +108,7 @@ pub struct VMLogic<'a> {
     events: Vec<Event>,
     actions: Vec<Vec<u8>>,
     root_hash: Option<[u8; 32]>,
+    proposals: Vec<Vec<u8>>,
```

Comment:
so once the ID has been assigned (by `VMHostFunctions::send_proposal`), it needs to be here too

so this will become a `Map<[u8; 32], Vec<u8>>`